test/Mustache/Test/ContextTest.php

```diff
@@ -180,6 +180,29 @@ public function testAnchoredDotNotationThrowsExceptions()
         $context->push(array('a' => 1));
         $context->findAnchoredDot('a');
     }
+
+    /**
+     * @expectedException Mustache_Exception_UnknownVariableException
+     */
+    public function testUnknownVariableThrowsException()
+    {
+        $context = new Mustache_Context(null, true);
+        $context->push(array('a' => 1));
+        $context->find('b');
+    }
+
+    /**
+     * @expectedException Mustache_Exception_UnknownVariableException
+     */
+    public function testAnchoredDotNotationUnknownVariableThrowsException()
+    {
+        $context = new Mustache_Context(null, true);
+        $a = array(
+            'a' => array('b' => 1),
+        );
+        $context->push($a);
+        $context->find('a.c');
+    }
```

Comment:
i'd like to see positive and negative tests. we can probably do it with a single method for "throws exception" and a single method for "doesn't throw exception" and a dataprovider for each.

for the examples below, assume context like this:

```php
[
  'a' => ['b' => 1],
  'c' => 1,
  'd' => 0,
]
```

we need coverage for:

- finding things that exist doesn't throw, with coverage for top level variables, variables in nested sections, dot notation, and top level variables from inside sections.
  e.g.: `{{ a }}`, `{{# a }}{{ b }}{{/ a }}`, `{{ a.b }}` and `{{# c }}{{ a }}{{/ c }}`, etc.

- sections and inverted sections never throw.
  e.g.: `{{# x }}{{/ x }}`, `{{^ x }}{{/ x }}`, `{{# a }}{{# x }}{{/ x }}{{/ a }}`, `{{# a.x }}{{/ a.x }}`, `{{# d.x }}{{/ d.x }}`, etc.

- missing variables inside falsey sections never throw.
  e.g.: `{{^ a }}{{ x }}{{/ a }}` and `{{# d }}{{ x }}{{/ d }}`, etc.